src/dos/cdrom_image.cpp

```diff
@@ -294,9 +308,12 @@ bool CDROM_Interface_Image::AudioFile::seek(const uint32_t requested_pos)
 		        elapsed_ms, average_cdrom_seek_ms);
 #endif
 
-	// Only store the track's new position if the seek was successful
+	// Seeking in audio file is an audio-task, so update our audio position
 	if (result)
-		track_pos = requested_pos;
+		audio_pos = requested_pos;
+	else
+		audio_pos = {};
```

Comment:
Why not `audio_pos = 0;`?

Reply:
I had read somewhere that `{}` resets class/struct members back to their original initialized values, so I had wanted this to be `std::numeric_limits<uint32_t>::max()`. 

But alas.. that's a myth and it simply sets it to zero, which makes your comment perfectly valid! (to answer why we don't want to save `0` in the failed-seek case: if this functions`requested_pos` argument was also `0` but we failed to seek to `0`, then saving `0` would cause subsequent streaming-decodes to assume the track is correctly positioned when it's actually not; and is probably in a broken state).

I've added this correct back into the main commit.